sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/Transforms.java

```diff
@@ -183,6 +185,43 @@ private static PageMetadata getPageInfo(ReadResult readResultItem) {
             readResultItem.getAngle(), DimensionUnit.fromString(readResultItem.getUnit().toString()));
     }
 
+    /**
+     * Automatically detect byte buffer's content type.
+     * Given the source: <a href="https://en.wikipedia.org/wiki/Magic_number_(programming)#Magic_numbers_in_files"/>.
+     *
+     * @param buffer The byte buffer input.
+     *
+     * @return The {@link ContentType} content type.
+     */
+    static ContentType getContentType(ByteBuffer buffer) {
+        final byte[] bytes = buffer.array();
+        if (bytes.length < 4) {
+            throw new IllegalArgumentException("Invalid input. Expect more than 4 bytes of data");
+        }
+
+        final int byte0 = Byte.toUnsignedInt(bytes[0]);
+        final int byte1 = Byte.toUnsignedInt(bytes[1]);
+        final int byte2 = Byte.toUnsignedInt(bytes[2]);
+        final int byte3 = Byte.toUnsignedInt(bytes[3]);
+
+        if (byte0 == Byte.toUnsignedInt((byte) 0x25) && byte1 == Byte.toUnsignedInt((byte) 0x50) &&
+            byte2 == Byte.toUnsignedInt((byte) 0x44) && byte3 == Byte.toUnsignedInt((byte) 0x46)) {
+            return ContentType.APPLICATION_PDF;
+        } else if (byte0 == Byte.toUnsignedInt((byte) 0xff) && byte1 == Byte.toUnsignedInt((byte) 0xd8)) {
```

Comment:
For Jpeg, I think you also need to check the last two bytes are 0xFF and 0xD9

Reply:
I removed this for python

Reply:
ok ignore this